src/vs/workbench/services/suggest/browser/simpleCompletionModel.ts

```diff
@@ -208,10 +208,33 @@ export class SimpleCompletionModel {
 				// Then by file extension length ascending
 				score = a.fileExtLow.length - b.fileExtLow.length;
 			}
-			if (score === 0 || fileExtScore(a.fileExtLow) === 0 && fileExtScore(b.fileExtLow) === 0) {
-				// both files or directories, sort alphabetically
-				score = a.completion.label.localeCompare(b.completion.label);
+			if (score === 0 || (fileExtScore(a.fileExtLow) === 0 && fileExtScore(b.fileExtLow) === 0)) {
+				// Both are files or directories
+
+				// Remove trailing slashes for comparison so that /dir/vscode/ is considered a prefix of /dir/vscode-copilot/
+				const labelA = a.completion.label.replace(/\/$/, '');
+				const labelB = b.completion.label.replace(/\/$/, '');
```

Comment:
This is a good case to add a precomputed value to the completion, like `labelLowExcludeTrailingSep`